### PR TITLE
Cooldown + Fix Command.kit Logic

### PR DESCRIPTION
**Pull request by valkyrienyanko**

### CODE REVIEW

File: src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt [hunk 1]
```diff
@@ -28,9 +29,12 @@ abstract class Command(val name: String) : CommandExecutor {
         for (permission in permissions)
             if (!sender.hasPermission(permission)) {
                 plugin raise PermissionException() send sender
-                return true
+                return false
```
Comment (aeuu) [inline src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt]: Please revert this change. Returning false will print usage information which is not desired.

File: src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt [hunk 2]
```diff
@@ -28,9 +29,12 @@ abstract class Command(val name: String) : CommandExecutor {
...
             }
 
+        if (Cooldown.active((sender as Player).uniqueId))
```
Comment (aeuu) [inline src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt]: `Sender` may not be a `Player`, which will cause an Exception at runtime

File: src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt [hunk 3]
```diff
@@ -28,9 +29,12 @@ abstract class Command(val name: String) : CommandExecutor {
...
+            return false
```
Comment (aeuu) [inline src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt]: return `true` and send an exception instead. Maybe `CooldownException : InternalPluginException()`

File: src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt [hunk 4]
```diff
@@ -40,7 +44,7 @@ abstract class Command(val name: String) : CommandExecutor {
                 return true
             }
 
-            if (sender is Player && annotation.target.contains(CommandTarget.Player)) {
+            if (sender !is ConsoleCommandSender && annotation.target.contains(CommandTarget.Player)) {
```
Comment (aeuu) [inline src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt]: What is wrong with the original logic?

File: src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt [hunk 5]
```diff
@@ -52,11 +56,11 @@ abstract class Command(val name: String) : CommandExecutor {
         }
 
         plugin raise ExecutionTargetException(
-            if (sender is Player) CommandTarget.Player
+            if (sender !is ConsoleCommandSender) CommandTarget.Player
```
Comment (aeuu) [inline src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt]: This is a valid change, however, you may need to edit the messages in `messages.yml` and the default in `ExecutionTargetException` to reach the intended effect.

File: src/main/kotlin/io/arct/arctlib/utils/Cooldown.kt [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.*
+
+/**
+ * Simulate a cooldown given a UUID and duration
+ *
+ * @param id The UUID assigned to this cooldown
+ * @param duration The duration of the cooldown in seconds
+ * @constructor Starts the cooldown
+ */
+class Cooldown(id: UUID, val duration: Int) {
```
Comment (aeuu) [inline src/main/kotlin/io/arct/arctlib/utils/Cooldown.kt]: This is kind of a weird pattern, because you create a new object (`Cooldown`) to create a cooldown. Maybe this should be an object instead that has a function that lets you create a cooldown instead?

**Review by aeuu — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt (hunk 4 above)
Comment (valkyrienyanko) [reply src/main/kotlin/io/arct/arctlib/plugin/commands/Command.kt]: IDE being dumb, I'll revert it

File: src/main/kotlin/io/arct/arctlib/utils/Cooldown.kt (hunk 6 above)
Comment (valkyrienyanko) [reply src/main/kotlin/io/arct/arctlib/utils/Cooldown.kt]: "Maybe this should be an object instead that has a function that lets you create a cooldown instead?"

No idea how to do this.

**Review by aeuu — APPROVED** [review]
